src/server/src/classes/Action.ts

```diff
@@ -0,0 +1,90 @@
+import { ActionType } from "../enums";
+import { IAction } from "../interfaces/IAction";
+import { IPlayer } from "../interfaces/IPlayer";
+import { IRound } from "../interfaces/IRound";
+
+/** Action:
+ *  - Performs any poker action
+ *  - Updates player & round state accordingly
+ */
+
+export class Action {
+  private action: IAction;
+  private player: IPlayer;
+  private round: IRound;
+
+  constructor(action: IAction, player: IPlayer, round: IRound) {
+    this.action = action;
+    this.player = player;
+    this.round = round;
+  }
+
+  performAction() {
+    switch (this.action.actionType as ActionType) {
```

Comment:
potential alternative is something like
`if (this.action.actionType in ActionType) { eval(this.action.actionType) }` but not sure if that would work/if it's secure

Reply:
I tried this but couldnt get it to work, maybe we could revisit later on